src/encoder.rs

```diff
@@ -1380,6 +1380,7 @@ pub struct StreamWriter<'a, W: Write> {
     writer: Wrapper<'a, W>,
     prev_buf: Vec<u8>,
     curr_buf: Vec<u8>,
+    filtered_buf: Vec<u8>,
```

Comment:
Wondering if the three buffers should be encoded as `Box<[u8]>` to ensure that the lengths are never mutated?

Then `vec![0; in_len].into_boxed_slice()` in the constructor.

Reply:
Rather not. Actually there should be length information associated with the line buffer. If we consider encoding of APNG frames we will have to change the length at runtime (soonish) for functionality anyways. I don't think the gains of a `usize` in type size warrant the readability change.